components/SatisfactionSlider.tsx

```diff
@@ -39,11 +39,22 @@ const SatisfactionSlider = ({
 
   return (
     <>
-      {title && <h3>{title}</h3>}
       <fieldset className={styles.wrapper} disabled={disabled}>
+        {title && <legend id={`${title}-legend`}>{title}</legend>}
```

Comment:
If we switch from `h3` to `legend` we should make sure similar changes exist across all components